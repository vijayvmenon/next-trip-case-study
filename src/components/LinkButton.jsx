import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { colors } from '../constants';

const useStyles = createUseStyles({
  linkButtonRoot: {
    textDecoration: 'none',
    padding: 16,
    background: colors.BACKGROUND_1,
    color: colors.LINK_COLOR,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '1.5rem',
    fontWeight: 500,
    position: 'relative',
    // "&:hover": {
    //   background: colors.BACKGROUND_3,
    //   color: colors.WHITE,
    // },
  },
  linkActive: {
    background: colors.BACKGROUND_2,
    color: colors.WHITE,
    borderRadius: 5,
    zIndex: 1,
    boxShadow: `0px -2px 5px -3px #0E0E0E, 
    -2px 0px 5px -3px #0E0E0E, 
    2px 0px 5px -3px #0E0E0E, 
    0px 2px 5px -3px #0E0E0E`,
  },
});

export default function LinkButton({ name, hrefLink, active, click }) {
  const classes = useStyles();

  const clickFn = (e) => {
    // e.preventDefault();
    console.log(e.target);
    click(name);
  };

  return (
    <a
      href={hrefLink}
      className={`${classes.linkButtonRoot} ${
        active ? classes.linkActive : undefined
      }`}
      onClick={clickFn}
    >
      {name}
    </a>
  );
}

LinkButton.propTypes = {
  name: PropTypes.string.isRequired,
  hrefLink: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  click: PropTypes.func.isRequired,
};
