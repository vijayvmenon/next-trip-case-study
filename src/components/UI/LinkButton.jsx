import React from 'react';
import PropTypes from 'prop-types';
import { Link, useLocation } from 'wouter';
import { createUseStyles } from 'react-jss';
import { colors } from '../../constants';

const useStyles = createUseStyles({
  linkButtonRoot: {
    textDecoration: 'none',
    padding: 16,
    background: colors.BACKGROUND_1,
    color: colors.LINK_COLOR,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '1.5rem',
    fontWeight: 500,
    position: 'relative',
    // "&:hover": {
    //   background: colors.BACKGROUND_3,
    //   color: colors.WHITE,
    // },
  },
  linkActive: {
    background: colors.BACKGROUND_2,
    color: colors.WHITE,
    borderRadius: 5,
    zIndex: 1,
    boxShadow: `0px -2px 5px -3px #0E0E0E, 
    -2px 0px 5px -3px #0E0E0E, 
    2px 0px 5px -3px #0E0E0E, 
    0px 2px 5px -3px #0E0E0E`,
    '&:after': {
      content: '""',
      display: 'block',
      width: 0,
      height: 0,
      position: 'absolute',
      borderTop: '12px solid #0071DE',
      borderLeft: '12px solid transparent',
      borderRight: '12px solid transparent',
      left: '45%',
      bottom: '-8px',
      filter: 'drop-shadow(1px 3px 1px rgb(0,0,0,0.2))',
      zIndex: 1,
    },
  },
});

export default function LinkButton({ name, hrefLink }) {
  const classes = useStyles();
  // const [match, params] = useRoute(`${hrefLink}?/:route/:direction/:stop*`); // check if route is the current active route
  const [location] = useLocation();
  const active = location.indexOf(hrefLink) !== -1;
  return (
    <Link
      className={`${classes.linkButtonRoot} ${
        active ? classes.linkActive : undefined
      }`}
      href={hrefLink}
      data-testid={
        active ? `custom-anchor-${name}-active` : `custom-anchor-${name}`
      }
    >
      {name}
    </Link>
  );
}
LinkButton.defaultProps = {
  name: '',
};

LinkButton.propTypes = {
  name: PropTypes.string,
  hrefLink: PropTypes.string.isRequired,
};
