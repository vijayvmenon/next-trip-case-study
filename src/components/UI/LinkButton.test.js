import { render, fireEvent, screen } from '@testing-library/react';
import LinkButton from './LinkButton';
import renderer from 'react-test-renderer';

const props = {
  hrefLink: '/test',
  name: 'test',
};

describe('Component:LinkButton.jsx', () => {
  test('LinkButton.jsx renders correctly', () => {
    const linkButton = renderer.create(<LinkButton {...props} />).toJSON();
    expect(linkButton).toMatchSnapshot();
  });

  test('LinkButton becomes active on triggering  a Click', () => {
    const { container, getByTestId } = render(<LinkButton {...props} />);
    fireEvent.click(getByTestId('custom-anchor-test'));
    expect(getByTestId('custom-anchor-test-active')).toBeTruthy();
  });
});
