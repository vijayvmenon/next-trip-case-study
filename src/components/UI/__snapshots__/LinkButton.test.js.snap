// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:LinkButton.jsx LinkButton.jsx renders correctly 1`] = `
<a
  className="linkButtonRoot-0-2-1 undefined"
  data-testid="custom-anchor-test"
  href="/test"
  onClick={[Function]}
  to={null}
>
  test
</a>
`;
